interpreter/golabels/runtime_data.go

```diff
@@ -0,0 +1,115 @@
+package golabels // import "go.opentelemetry.io/ebpf-profiler/interpreter/golabels"
+
+// #include <stdlib.h>
+// #include "../../support/ebpf/types.h"
+import "C"
+
+// defaultVersion is used if the go binary has an unrecognized major+minor version.
+// Consider bumping this whenever a new version of Go is released.
+var latestVersion = "go1.24"
```

Comment:
I would love to drop this and its related check.

This aligns also with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/409/files#r2035872643 - a simple switch on the version should be preferred.

Sketching something along this idea (no valid syntax!):
```
offsets := GoCustomLabelsOffsets{
		m_offset:               48,
		curg:                   192,
		labels:                 344,
		hmap_count:             0,
		hmap_log2_bucket_count: 9,
		hmap_buckets:           16,
}
switch goVersion {
case go1.13,go1.14, ..:
 // the default offsets are just fine
case go1.21, go.122:
 // apply changes to offset
 offsets.labels = 344
case go1.24
 fallthrough
default:
 offsets.labels = 352 
}

```